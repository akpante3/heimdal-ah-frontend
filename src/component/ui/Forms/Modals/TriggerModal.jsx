import React, { PureComponent, Fragment } from 'react';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';
import Button from '../../buttons/Button';
import './TriggerModal.scss';

class TriggerModal extends PureComponent {
  state = {
    modal: false,
  };

  toggle = () => {
    this.setState(prevState => ({
      modal: !prevState.modal,
    }));
  };

  render() {
    return (
      <Fragment>
        <Button
          className="header-button"
          type={this.props.type}
          label={this.props.label}
          onClick={this.toggle}
        />
        <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle} />
          <ModalBody>{this.props.children}</ModalBody>
        </Modal>
      </Fragment>
    );
  }
}

export default TriggerModal;
