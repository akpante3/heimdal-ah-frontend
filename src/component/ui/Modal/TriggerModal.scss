.modal-header {
  border-bottom: none;
}

.modal-link {
 background-color: antiquewhite
}
