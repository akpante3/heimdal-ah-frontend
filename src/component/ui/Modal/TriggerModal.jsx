import React, { Fragment, Component } from 'react';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';
import './TriggerModal.scss';
import Button from '../Buttons/Button';

/**
 * @description - Heimdal Authors TriggerModal Component
 * @param {props} Class - the class of the button(bootstrap or custom)
 * @param {function} toggle - the function to be performed when the modal is triggered
 * @returns {fragment} Modal fragment
 */
class TriggerModal extends Component {
  state = {
    modal: false,
  };

  toggle = () => {
    this.setState(prevState => ({
      modal: !prevState.modal,
    }));
  };

  render() {
    return (
      <Fragment>
        <Button
          className="header-button"
          type={this.props.type}
          label={this.props.label}
          onClick={this.toggle}
        />
        <Modal
          isOpen={this.state.modal}
          toggle={this.toggle}
          className={this.props.className}
        >
          <ModalHeader toggle={this.toggle} />
          <ModalBody>{this.props.children}</ModalBody>
        </Modal>
      </Fragment>
    );
  }
}

export default TriggerModal;
