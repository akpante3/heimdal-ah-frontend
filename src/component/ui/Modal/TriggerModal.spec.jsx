/* eslint-disable no-undef */
import React from 'react';
import toJson from 'enzyme-to-json';
import TriggerModal from './TriggerModal';

describe('Render <TriggerModal />', () => {
  it('it should render TriggerModal component', () => {
    const container = shallow(<TriggerModal />);
    expect(container.length).toBe(1);
  });

  it('should match the snapshot', () => {
    const tree = shallow(<TriggerModal />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
