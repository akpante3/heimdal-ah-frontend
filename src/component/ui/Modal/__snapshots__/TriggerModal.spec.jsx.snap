// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <ModalButton /> should match the snapshot 1`] = `
<Fragment>
  <Button
    Class=""
    label="Default"
    onClick={[Function]}
    type="default"
  />
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    />
    <ModalBody
      tag="div"
    />
  </Modal>
</Fragment>
`;
